lightning/src/routing/router.rs

```diff
@@ -1013,8 +1021,17 @@ pub fn get_route<L: Deref>(our_node_id: &PublicKey, network: &NetworkGraph, paye
 				let mut ordered_hops = vec!((new_entry.clone(), NodeFeatures::empty()));
 
 				'path_walk: loop {
-					if let Some(&(_, _, _, ref features)) = first_hop_targets.get(&ordered_hops.last().unwrap().0.pubkey) {
-						ordered_hops.last_mut().unwrap().1 = features.clone();
+					let mut features_set = false;
+					if let Some(first_channels) = first_hop_targets.get(&ordered_hops.last().unwrap().0.pubkey) {
+						for (scid, _, _, ref features) in first_channels {
+							if *scid == ordered_hops.last().unwrap().0.short_channel_id {
+								ordered_hops.last_mut().unwrap().1 = features.clone();
+								features_set = true;
+								break;
+							}
+						}
+					}
+					if features_set {
 					} else if let Some(node) = network_nodes.get(&ordered_hops.last().unwrap().0.pubkey) {
```

Comment:
I mean easier than rewriting the rest of the code?

Reply:
This makes me less uncomfortable and passes all tests:
```
diff --git a/lightning/src/routing/router.rs b/lightning/src/routing/router.rs
index 38dc918d..3bfab5a9 100644
--- a/lightning/src/routing/router.rs
+++ b/lightning/src/routing/router.rs
@@ -1031,21 +1031,21 @@ pub fn get_route<L: Deref>(our_node_id: &PublicKey, network: &NetworkGraph, paye
                                                        }
                                                }
                                        }
-                                       if features_set {
-                                       } else if let Some(node) = network_nodes.get(&ordered_hops.last().unwrap().0.pubkey) {
-                                               if let Some(node_info) = node.announcement_info.as_ref() {
-                                                       ordered_hops.last_mut().unwrap().1 = node_info.features.clone();
+                                       if !features_set {
+                                               if let Some(node) = network_nodes.get(&ordered_hops.last().unwrap().0.pubkey) {
+                                                       if let Some(node_info) = node.announcement_info.as_ref() {
+                                                               ordered_hops.last_mut().unwrap().1 = node_info.features.clone();
+                                                       } else {
+                                                               ordered_hops.last_mut().unwrap().1 = NodeFeatures::empty();
+                                                       }
                                                } else {
-                                                       ordered_hops.last_mut().unwrap().1 = NodeFeatures::empty();
+                                                       // We should be able to fill in features for everything except the last
+                                                       // hop, if the last hop was provided via a BOLT 11 invoice (though we
+                                                       // should be able to extend it further as BOLT 11 does have feature
+                                                       // flags for the last hop node itself).
+                                                       assert!(ordered_hops.last().unwrap().0.pubkey == *payee);
                                                }
-                                       } else {
-                                               // We should be able to fill in features for everything except the last
-                                               // hop, if the last hop was provided via a BOLT 11 invoice (though we
-                                               // should be able to extend it further as BOLT 11 does have feature
-                                               // flags for the last hop node itself).
-                                               assert!(ordered_hops.last().unwrap().0.pubkey == *payee);
                                        }
-
                                        // Means we succesfully traversed from the payer to the payee, now
                                        // save this path for the payment route. Also, update the liquidity
                                        // remaining on the used hops, so that we take them into account
```